PR: https://github.com/odoo/odoo/pull/

From: 665781d5a8524e73ce2649665d1113d9e6c854d6
From: Yannick Tivisse
Date: 2016-09-02 11:44:15

Structural Changes: 6
Total Changes: 8

[IMP] various: Move all configuration fields from res_company form to the related modules

=======
Purpose
=======

The company form should be heavily simplified. It's complex to have some settings on the company form, and others on the Settings menu of the related app. It would be much easier to have all settings in Settings menu (res.config) and nothing on the company, even if some of the Settings are multi company. (Stored on the company but set from the Settings menu).

================================= pseudo patch: =================================

--- a/addons/purchase/models/res_config.py
+++ b/addons/purchase/models/res_config.py
@@ -8,6 +8,14 @@ class PurchaseConfigSettings(models.TransientModel):
     _name = 'purchase.config.settings'
     _inherit = 'res.config.settings'
 
+    company_id = fields.Many2one('res.company', string='Company', required=True,
+        default=lambda self: self.env.user.company_id)
+    po_lead = fields.Float(related='company_id.po_lead', string="Purchase Lead Time *")
+    po_lock = fields.Selection(related='company_id.po_lock', string="Purchase Order Modification *")
+    po_double_validation = fields.Selection(related='company_id.po_double_validation', string="Levels of Approvals *")
+    po_double_validation_amount = fields.Monetary(related='company_id.po_double_validation_amount', string="Double validation amount *", currency_field='company_currency_id')
+    company_currency_id = fields.Many2one('res.currency', related='company_id.currency_id', readonly=True,
+        help='Utility field to express amount currency')
     group_product_variant = fields.Selection([
         (0, "No variants on products"),
         (1, 'Products can have several attributes, defining variants (Example: size, color,...)')
